### PR TITLE
Interactive version of plot_evoked_fieldmap

### PR DESCRIPTION
**Pull request by wmvanvliet**
Continuation of #11676  but now on top of the new UI events system. This PR is built on top of #11891

All the functionality is there I think. Now it just needs to be made a bit more pretty.

Todo:

 - [x] Implement the core functionality
 - [x] Make HTML notebook version pretty
 - [x] Move time interaction to `renderer` so it can be shared between `Brain` and `EvokedField`
 - [x] Add unit tests
 - [x] Add documentation
 - [x] Update What's new

### CODE REVIEW

**Comment by wmvanvliet** [comment]
With this version, you can either plot the evoked fields in a new figure, while enjoying being able to scroll through time. Or, you can plot the evoked fields on top of an existing `Brain` figure, call `ui_events.link` on them, and watch it integrate smoothly.

**Comment by larsoner** [comment]
@wmvanvliet let me know when I should look

**Comment by wmvanvliet** [comment]
It's looking pretty good, but it's not completely there yet (I've updated the TODO list). Struggling with the HTML notebook code.

**Comment by wmvanvliet** [comment]
In this PR I made a big change: the time interaction code moved to `renderer` so other figures can also have time scrolling and animation and such, not just `Brain`. Do you agree with this choice and the way it's implemented currently?

**Comment by larsoner** [comment]
> Do you agree with this choice and the way it's implemented currently?

My initial reaction was that I wasn't sure how you would do it, but then I remembered the backend abstraction layer. So yes if you can do it this way and make it look okay for both Brain (which maybe should look the same as in `main`?) and this new EvokedField then that's great!

From a quick look at the code, it looks like to do this you more or less moved a bunch of code from `Brain` to the abstraction layer, I'm assuming with minor adjustments. Then EvokedField is a port/copy-paste of what used to be in `mne/viz/_3d.py` but with more extensive adjustments. Does that sound right from a high level perspective?

**Comment by wmvanvliet** [comment]
exactly right.

**Comment by wmvanvliet** [comment]
Scrolling through time in the notebook is still a bit janky due to how the animation system interacts with the UI events, but other than that, if anyone can spot anything broken/missing/weird I'd love to hear about it.

**Comment by larsoner** [comment]
CIs are not happy

Ready for review/merge from your end?

FWIW a lot of folks I work with still use Xfit for dipole fitting. It feels like with this system in place we might "just" be a week's worth of (mostly GUI) work away from being able to replace it....

**Comment by wmvanvliet** [comment]
We need some more unit tests (make CIs even more unhappy) and there are still a few lambda functions that should be `@_auto_weakref`.

**Comment by wmvanvliet** [comment]
> FWIW a lot of folks I work with still use Xfit for dipole fitting. It feels like with this system in place we might "just" be a week's worth of (mostly GUI) work away from being able to replace it....

And that's my code sprint plan :)

**Comment by larsoner** [comment]
notebook will fail until I fix it in https://github.com/mne-tools/mne-python/pull/11982 so just ignore it

**Comment by larsoner** [comment]
... and can you also touch these examples/tutorials so that we can verify in CircleCI that they look at least as good as before:
```
$ git grep "plot_\(evoked_\)\?field" | grep "\(tutorials\|examples\)/.*"
examples/datasets/spm_faces_dataset_sgskip.py:evoked[0].plot_field(maps, time=0.170)
examples/visualization/mne_helmet.py:evoked.plot_field(maps, time=time, fig=fig, time_label=None, vmax=5e-13)
tutorials/evoked/20_visualize_evoked.py:# Once the maps are computed, you can plot them with `evoked.plot_field()
...
```

**Comment by wmvanvliet** [comment]
Green! :star_struck:

Ok, now I can add more unit tests and see if I can fix the time scrolling behavior on notebooks.

**Comment by wmvanvliet** [comment]
While digging in the whole playback business I stumbled across the `_PlayMenu` class that is there but doesn't seem to be used at all. Looks like stuff we wanted to implement at some point but abandoned.

**Comment by larsoner** [comment]
Feel free to delete it if it's not used, it's in `git` history if we need it later. And we might want to implement it differently/better in the end anyway

**Comment by wmvanvliet** [comment]
Ready from my end!

**Comment by larsoner** [comment]
CircleCI failed, feel free to look or I can look later

**Comment by larsoner** [comment]
CIs should be fixed, will test and look at code hopefully tomorrow!

File: mne/viz/evoked_field.py [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+                        kind=f"field_strength_{type}",
+                        fmin=-vmax,
+                        fmax=vmax,
+                    ),
+                )
+            break
+        else:
+            raise ValueError(f"No {type.upper()} field map currently shown.")
+
+    def rescale(self):
```
Comment (larsoner) [inline mne/viz/evoked_field.py:571]: should be private I think

File: mne/viz/tests/test_3d.py [hunk 2]
```diff
@@ -185,6 +187,21 @@ def test_plot_evoked_field(renderer):
             )
         evoked.plot_field(maps, time=0.1, n_contours=n_contours)
 
+    # Test plotting inside an existing Brain figure. This should not work in a notebook.
+    brain = Brain("fsaverage", "lh", "inflated", subjects_dir=subjects_dir)
+    if get_3d_backend() == "notebook":
```
Comment (larsoner) [inline mne/viz/tests/test_3d.py:192]: I thought that `renderer` ran both notebook and non-notebook variants, but it doesn't look that way :(

Looks like we'd have to create a `test_notebook.py` like this to test this stuff properly on notebook, basically each "test" contains Python code that must be self-contained (with all imports and variables etc.) necessary to run it in `nbexec`:

https://github.com/mne-tools/mne-python/blob/main/mne/viz/_brain/tests/test_notebook.py

One fairly low-effort solution is to write some minimal code to take care of covering the `time_viewer=True` and `time_viewer=False` cases plus the Brain-embedding case (which would cover this `NotImplementedError`. If you're sick of this notebook business I could push it quickly I think.

**Review by larsoner — COMMENTED** [review]
When I run the example and slide the Brain time slider I get updates to the clim sliders that are incorrect (they change which they shouldn't, *and* they don't update the clim values actually), can you replicate?

Screenshot 2023-09-12 at 5 09 07 PM

But other than that it's remarkably smooth!

**Comment by wmvanvliet** [comment]
> When I run the example and slide the Brain time slider I get updates to the clim sliders that are incorrect (they change which they shouldn't, and they don't update the clim values actually), can you replicate?

Unfortunately, this doesn't replicate on my Window nor my Linux machine. Is this on OSX?

File: mne/dipole.py [hunk 3]
```diff
@@ -818,7 +818,7 @@ def _read_dipole_bdip(fname):
                 has_errors = this_has_errors
                 for key in _BDIP_ERROR_KEYS:
                     conf[key] = list()
-            assert has_errors == this_has_errors
+            # assert has_errors == this_has_errors
```
[the remark this answers is not in the document]
Comment (wmvanvliet) [reply mne/dipole.py:821]: ah yes, I have a weird old dipole file that only loads when I comment this out and it accidentally got mixed into this PR. Reverting this.

File: mne/viz/_3d.py [hunk 4]
```diff
@@ -417,150 +419,74 @@ def plot_evoked_field(
...
+        ``True`` for the PyVista backend and ``False`` otherwise.
+
+        .. versionadded:: 1.6
     %(verbose)s
 
     Returns
     -------
-    fig : instance of Figure3D
-        The figure.
+    fig : mne.viz.EvokedField
```
[the remark this answers is not in the document]
Comment (wmvanvliet) [reply mne/viz/_3d.py:470]: This is a good idea!

File: mne/viz/evoked_field.py (hunk 1 above)
Comment (wmvanvliet) [reply mne/viz/evoked_field.py:571]: yeah I guess so.

File: mne/viz/evoked_field.py [hunk 5]
```diff
@@ -0,0 +1,576 @@
...
+        subscribe(self, "contours", self._on_contours)
+
+        if not self._in_brain_figure:
+            self._renderer.set_camera(azimuth=10, elevation=60)
+
+        self._renderer.show()
+
+    def _prepare_surf_map(self, surf_map, color, alpha):
+        """Compute all the data required to render a fieldlines map."""
+        from scipy.interpolate import interp1d
```
[the remark this answers is not in the document]
Comment (wmvanvliet) [reply mne/viz/evoked_field.py:249]: cool!

**Comment by larsoner** [comment]
> Unfortunately, this doesn't replicate on my Window nor my Linux machine. Is this on OSX?

I was, but after updating I no longer see this behavior. Not sure if you fixed something or I had some broken code, but I think we're okay now!

CircleCI and pip-pre are being fixed in https://github.com/mne-tools/mne-python/pull/11926 which I'll merge into this branch once it's in. I'll also look at the code again and mark for green when merge assuming I don't find anything else, thanks in advance!

**Comment by larsoner** [comment]
@wmvanvliet can you quickly fix the failures here?

https://app.circleci.com/pipelines/github/mne-tools/mne-python/20847/workflows/308d8d33-9678-42e4-9dcb-51f2b2ec3d08/jobs/59285

If not I can do it. I think we maybe just need `time_viewer=False` or so

**Comment by wmvanvliet** [comment]
https://github.com/mne-tools/mne-python/pull/12005

**Pull request merged**